docs/src/apalache/assignments.md

```diff
@@ -0,0 +1,233 @@
+What makes it different from action `A` is the fact that this is far from immediately obvious from the syntax. 
+The fact that there happened to be a closed-form solution for which gave us an integer value for `y'`, is a lucky coincidence, as `B` could have been, for example, `y' * y' + 1 = 0`, with no real roots. 
+To avoid cases like this, we require that transition operators explicitly declare the values of state variables in successor states.
+
+We call syntactic forms, which explicitly represent successor state values, _assignment candidates_. An assignment candidate for `x` is a TLA+ expression that has one of the following forms:
+- `x' = e`, 
+- `x' \in S`,
+- `UNCHANGED x`, or
+- `x' := e` (note that `:=` is the operator defined in `Apalache.tla`)
+
+So to reformulate the second requirement: the transition operator must contain at least one assignment candidate for each variable declared in the specification.
+
+### Control Flow: Minimal and Compound Actions
+When writing non-trivial specifications, authors often end up with something similar to the following:
+```tla
+EventA == ...
+EventB == ...
+...
+
+Next == \/ EventA
+        \/ EventB
+```
+Specifically, `EventA` and `EventB` often represent mutually exclusive possibilities of execution.
+Just like before, the basic definition of an action is not sufficient to explain the relation of `EventA` or `EventB` and `Next`;
+if `EventA` is an action and `EventB` is an action, then `Next` is also an action.
+To more accurately describe this scenario, we observe that the operator or ( `\/`) sometimes serves as a kind of parallel composition operator (`||`) in process algebra - it connects two (or more) actions into a larger one.
+
+There are only two operators in TLA+ that could be considered control-flow operators in this way, the or (`\/`) operator and the if-then-else operator.
+We distinguish their uses as action- and as value operators:
+```tla
+
+A == x = 1 \/ x = 2 (* arguments are not actions *)
+B == x' = 1 \/ x' = 2 (* arguments are actions *)
+```
+Simply put, if all arguments to an operator `\/` are actions, then that operator is an action-or, otherwise it is a value-or. Similarly, if both the `THEN _` and `ELSE _` subexpressions of if-then-else are actions, it is an action-ITE, otherwise it is a value-ITE (in particular, a value-ITE can be non-Boolean).
+
+Using these two operators we can define the following terms:
+A _minimal action_ is an action which contains no action-or and no action-ITE.
+Conversely, a _compound action_ is an action which contains at least one action-or or at least one action-ITE.
+
+### Slices
+Given a transition operator, which is most commonly a compound action, we can decompose it into as many minimal actions as possible. We call this process _slicing_ and the resulting minimal actions _slices_.
```

Comment:
We don't decompose it into "as many actions as possible". We decompose it into an equivalent set of minimal actions. And in practice it is a rather small set. Otherwise, we could blast Next by computing DNF.

Reply:
"As many as possible" and "equivalent set of" aren't mutually exclusive. And neither are "as many as possible" and "few" (re: "small set")